include/zephyr/devicetree.h

```diff
@@ -1712,6 +1712,70 @@
  * @}
  */
 
+/**
+ * @defgroup devicetree-generic-vendor Vendor name helpers
+ * @ingroup devicetree
+ * @{
+ */
+
+/**
+ * @brief Is "idx" a valid vendor index?
+ *
+ * If this returns 1, then DT_NODE_VENDOR_BY_IDX(node_id, idx) is valid. If it
+ * returns 0, it is an error to use DT_NODE_VENDOR_BY_IDX(node_id, idx) with
+ * index "idx".
+ *
+ * @param node_id node identifier
+ * @param idx index of the vendor to check
+ * @return 1 if "idx" is a valid vendor index,
+ *         0 otherwise.
+ */
+#define DT_NODE_VENDOR_HAS_IDX(node_id, idx) \
+	IS_ENABLED(DT_CAT4(node_id, _COMPAT_VENDOR_IDX_, idx, _EXISTS))
+
+/**
+ * @brief Get the vendor at index "idx" as a string literal
+ *
```

Comment:
Nit pick, I'd move the `DT_NODE_VENDOR_BY_IDX` first as its the 'main' macro in this family of macros.

Can you add a bit more description on what `vendor` means here.  (ie, that we are mapping the 'vendor' portion of the compatible to a fuller string based on vendor-prefixes.txt

Reply:
Having an example of what this would return would be good.